### PR TITLE
Dropdown lists

### PR DESCRIPTION
**Pull request by sandrineBeauche**

### CODE REVIEW

File: scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/client/JobOutput.java [hunk 1]
```diff
@@ -81,48 +100,152 @@ public int compare(Task o1, Task o2) {
...
         this.jobId = jobId;
         this.lines = new TreeMap<Task, List<String>>(TASK_FINISHED_TIME_COMPARATOR);
     }
+    
+    
+    public List<String> update(Task task, String out, OutputMode outputMode){
+        ArrayList<String> remoteVisuLines = new ArrayList<String>();
 
+        
+        if(this.outputMode != outputMode){
```
Comment (lpellegr) [inline scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/client/JobOutput.java]: Problem with formatting sometimes there is a space before brackets and curly braces and sometimes not.

File: scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/client/controller/AbstractSelectedTargetController.java [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    public abstract void changeJobOutputContext(Job job);
+    
+    
+    public abstract void changeTaskOutputContext(Task task);
+    
+    
+    public String getNoTargetLabelContent(){
+        SelectionTarget target = this.model.getSelectionTarget();
+        switch(target){
+        case TASK_TARGET:
```
Comment (lpellegr) [inline scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/client/controller/AbstractSelectedTargetController.java]: The label could be part of a field defined in SelectionTarget enum. This way, the expected value could be retrieved with a call like  `this.model.getSelectionTarget().getNoTargetLabelContent()`, thus avoiding the _switch_.

File: scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/client/controller/OutputController.java [hunk 3]
```diff
@@ -254,66 +289,105 @@ public void onFailure(Throwable caught) {
...
-        return parentController;
+    
+    
+    
+    public Collection<List<String>> getLinesToDisplay(JobOutput output){
+        if(this.model.getSelectionTarget() == SelectionTarget.TASK_TARGET){
+            Task task = this.parentController.getSelectedTask();
+            List<String> lines = output.getLines(task);
+            
+            Collection<List<String>> result = new ArrayList<List<String>>();
```
Comment (lpellegr) [inline scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/client/controller/OutputController.java]: Could be initialized with capacity.

File: scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/client/controller/ResultController.java [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+            this.model.setDownloadOption(DownloadOption.OPT_BIN);
+        }
+    }
+    
+    
+    public void doDownload(DynamicForm form){
+        Task task = this.parentController.getSelectedTask();
+        if(task != null){
+            String taskId = task.getName();
+            form.getField("taskId").setValue(taskId);
```
Comment (lpellegr) [inline scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/client/controller/ResultController.java]: I think field names should be extracted in a constants and this constant used to reference the fieldname in all classes. This way, if the name change, no complex code modification is required. Besides, it is easier to tracker what is using it.

File: scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/client/SelectionTarget.java [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+    JOB_TARGET("Selected job");
+    
+    public final String label;
+    
+    private SelectionTarget(String label){
+        this.label = label;
+    }
+    
+    
+    public static String [] getValuesString(){
```
Comment (lpellegr) [inline scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/client/SelectionTarget.java]: I would have rather called it `toArray()` or `toStringArray`.

File: scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/client/controller/OutputController.java (hunk 3 above)
Comment (lpellegr) [inline scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/client/controller/OutputController.java]: Initial capacity could be set.

File: scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/client/view/ResultView.java [hunk 6]
```diff
@@ -79,39 +72,49 @@
...
-
-    private Layout root = null;
+public class ResultView implements TaskSelectedListener, JobOutputListener {
 
-    private SelectItem taskSelect = null;
-
-    private SchedulerController controller = null;
-
-    private Label label = null;
+    public static final String taskIdFieldName = "taskId";
```
Comment (lpellegr) [inline scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/client/view/ResultView.java]: Constants should use snake case and be written in uppercase.

**Comment by lpellegr** [comment]
Only one remaining comment for me regarding constants in `ResultView` which should be named using snake case and in uppercase. This last point can probably be done once the pull-request is merged.

File: scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/client/JobOutput.java [hunk 7]
```diff
@@ -81,48 +100,152 @@ public int compare(Task o1, Task o2) {
...
-    public void update(Task task, List<String> output) {
-        List<String> tl = this.lines.get(task);
-        if (tl != null) {
-            tl.clear();
-        } else {
-            tl = new ArrayList<String>();
-            this.lines.put(task, tl);
+    public List<String> updateLive(String out) {
+        ArrayList<String> remoteVisuLines = new ArrayList<String>();
+
```
Comment (fviale) [inline scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/client/JobOutput.java]: remoteVisuLines could be initialized using the string array length

**Pull request merged**